spec/design_topics/device_support.md

```diff
@@ -2,5 +2,76 @@
 
 # Device support
 
-TODO. See https://github.com/data-apis/array-api/issues/39
+For libraries that support execution on more than a single hardware device - e.g. CPU and GPU, or multiple GPUs - it is important to be able to control on which device newly created arrays get placed and where execution happens. Attempting to be fully implicit doesn't scale to situation with multiple GPUs.
 
+Existing libraries employ one or more of these three methods to exert such control:
+1. A global default device, which may be fixed or user-switchable.
+2. A context manager to control device assignment within its scope.
+3. Local control via explicit keywords and a method to transfer arrays to another device.
+
+This standard chooses to add support for method 3 (local control), because it's the most explicit and granular, with its only downside being verbosity. A context manager may be added in the future - see {ref}`device-out-of-scope` for details.
+
+## Syntax for device assignment
+
+The array API will offer the following syntax for device assignment and cross-device data transfer:
+
+1. A string representation to identify a device: `'device_type:index'`, with
+   `:index'` optional (e.g. doesn't apply to `'cpu'`). All lower-case, with type
+   strings `'cpu'`, `'gpu'`, `'tpu'`.
+2. A `device` object, whose constructor takes the string representation, with properties:
+   - `str`: the string representation.
+   - `type`: the device type part of the string representation.
+   - `index`: the device index, as an integer (with the first device of a given type having index `0`).
```

Comment:
Note that `str` and `type` are builtins, but they're the best descriptive names here I think. And PyTorch already has `type` and `index`, so on reflection that seemed more important than trying to use, e.g. `kind` and `id`.

Reply:
The spec already shadows a ton of builtin names (abs, min, max, ...). And it's more problematic for these ones because they are module-level names where users would typically import the names into the global scope. For this, it's an object where the names are attributes and don't actually conflict with builtin globals in user code.